tests/test_client.py

```diff
@@ -0,0 +1,237 @@
+            _ = client.fetch(CLIENT_SUPERMAN_URL)
+
+    @httpretty.activate
+    def test_fetch_retry_after(self):
+        """Test whether calls returning 503, 413, 429 status codes are retried"""
+
+        retry_after_value = 1
+
+        httpretty.register_uri(httpretty.GET,
+                               CLIENT_SPIDERMAN_URL,
+                               body="",
+                               status=413,
+                               forcing_headers={
+                                   'Retry-After': str(retry_after_value)
+                               })
+
+        httpretty.register_uri(httpretty.GET,
+                               CLIENT_SUPERMAN_URL,
+                               body="",
+                               status=429,
+                               forcing_headers={
+                                   'Retry-After': str(retry_after_value)
+                               })
+
+        httpretty.register_uri(httpretty.GET,
+                               CLIENT_BATMAN_URL,
+                               body="",
+                               status=503,
+                               forcing_headers={
+                                   'Retry-After': str(retry_after_value)
+                               })
+
+        client = MockedClient(CLIENT_API_URL)
+
+        urls = [CLIENT_SPIDERMAN_URL, CLIENT_SUPERMAN_URL, CLIENT_BATMAN_URL]
+
+        for url in urls:
+            before = int(time.time())
+            expected = before + (retry_after_value * client.max_retries)
+
+            with self.assertRaises(requests.exceptions.HTTPError):
+                _ = client.fetch(url)
+
+            after = int(time.time())
+            self.assertTrue(expected <= after)
+
+    @httpretty.activate
+    def test_fetch_retry(self):
+        """Test whether calls returning redirect status codes and 408, 423, 504 status codes are retried"""
+
+        httpretty.register_uri(httpretty.GET,
+                               CLIENT_IRONMAN_URL,
+                               body="",
+                               status=301)
+
+        httpretty.register_uri(httpretty.GET,
+                               CLIENT_SPIDERMAN_URL,
+                               body="",
+                               status=408)
+
+        httpretty.register_uri(httpretty.GET,
+                               CLIENT_SUPERMAN_URL,
+                               body="",
+                               status=423)
+
+        httpretty.register_uri(httpretty.GET,
+                               CLIENT_BATMAN_URL,
+                               body="",
+                               status=504)
+
+        client = MockedClient(CLIENT_API_URL)
+
+        urls = [CLIENT_SPIDERMAN_URL, CLIENT_SUPERMAN_URL, CLIENT_BATMAN_URL]
+
+        for url in urls:
+            with self.assertRaises(requests.exceptions.RetryError):
+                _ = client.fetch(url)
+
+    @httpretty.activate
+    def test_init_rate_limit(self):
```

Comment:
Rate limit these tests should go in a different test class. I also miss some tests here.